datastream/samplers.py

```diff
@@ -24,7 +24,7 @@ def __init__(self, length, proportion=1.0, replacement=False):
             replacement=replacement,
             sampler=torch.utils.data.WeightedRandomSampler(
                 torch.ones(length).double(),
-                num_samples=int(length * proportion),
+                num_samples=max(int(length * proportion), 1),
```

Comment:
Ties back to the earlier comment about zero lengths etc
Not sure how everything reacts, but usually things are more stable when even degenerate cases makes sense and works

Reply:
We can try to handle empty datastreams better. Not sure how easy it would be

Reply:
If its not supported then it should crash early, so crash on init with `ValueError` for example

Reply:
Added check